Jetbrains cursorless support

asoee opened the pull request:
This add support for cursorless in jetbrains, by adding a new package, where the output bundle can be embedded in the jetbrains plugin. (Currently maintained here: (https://github.com/asoee/cursorless-jetbrains))
It currently support most of the core functionality of cursorless: hats, selection, edits, flashing, treesitter, and some navigation commands.

It does not (yet) support:
- highlights
- tutorial
- cheatsheet

Caveats:
- It currently relies on manual cop/paste of the output bundle to the jetbrains plugin.
- tresitter support requires manual copy/paste of wasm/configuration files from the VSCode extension.

## Checklist

- [ ] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [ ] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [ ] I have not broken the cheatsheet

asoee committed:
quickjs build - take 1

asoee committed:
wip

asoee committed:
add more jetbrains skeleton

asoee committed:
wip

asoee committed:
wip

asoee committed:
first working hats with javet

asoee committed:
add setSelection callback

asoee committed:
add missing files

asoee committed:
fix jetbrains texteditor visible range
add multi editor support to jetbrains ide

asoee committed:
fix active editor state
update existing editors instead of recreating

asoee committed:
add clipboard support
make configuration injectable

asoee committed:
remove editor state on close

asoee committed:
remove debug logging

asoee committed:
add most of range IDE commands

asoee committed:
add reveal range

asoee committed:
add jetbrains treesitter support

asoee committed:
Merge branch 'cursorless-dev:main' into cursorless-jetbrains

asoee committed:
Remove neovim references from jetbrains package

asoee committed:
update pnpm-lock.yaml

asoee committed:
fix various linting issues

asoee committed:
fix more linting issues, and missing lodash dependency

asoee committed:
dont break build

asoee committed:
add path shim for cross os path parsing

asoee committed:
fix windows was path building

asoee committed:
add missing pathjoin file

asoee committed:
support editor visibility

asoee committed:
implement flashranges for jetbrains

asoee committed:
add support for prePhraseVersion

pokey commented:
@asoee would it be helpful for me to have a look at this point? Also, would be useful to have a quick blurb in description to indicate what's supported by this PR. Eg I'm assuming this PR doesn't support tree-sitter?

asoee committed:
enable multiple hat styles

asoee commented:
@pokey yes, please take a look.
I think it is slowly stabilizing now.
While the plugin does support tree-sitter. it is still the manual copy/paste way, and i have not looked into the treesitter-changes we discussed. Focus have been on making the plugin usable.

asoee committed:
inject colors and shape settings from ide

asoee committed:
jetbrains: use "go to declaration" for "follow" command

asoee committed:
improve editable editor support - add editable state

asoee committed:
Merge branch 'cursorless-dev:main' into cursorless-jetbrains

asoee committed:
add cursorless tag for jetbrains app

asoee committed:
Merge remote-tracking branch 'origin/main' into cursorless-jetbrains

asoee committed:
update to newest cursorless
reduce logging.

asoee committed:
change default runmode to production

asoee committed:
Merge remote-tracking branch 'origin/main' into cursorless-jetbrains

# Conflicts:
#	pnpm-lock.yaml

asoee committed:
map openlink to go to declaration

asoee committed:
compile fixes after merge from main

asoee committed:
esbuild fixes

asoee committed:
support selecting in readonly documents

asoee committed:
use new Query contructor for treesitter

phillco committed:
Add setHighlightRanges support for JetBrains IDE (#1)

Implements the setHighlightRanges method to enable persistent highlighting
functionality in the JetBrains Cursorless integration. This allows the
highlight command to work properly.

- Add setHighlightRanges to JetbrainsClient interface
- Implement the method in JetbrainsIDE class
- Pass highlightId, editorId, and serialized ranges to the client

asoee committed:
insert snippet support

asoee committed:
Merge remote-tracking branch 'origin/main' into cursorless-jetbrains

# Conflicts:
#	pnpm-lock.yaml

asoee committed:
mise config

asoee committed:
Merge remote-tracking branch 'refs/remotes/origin/main' into cursorless-jetbrains

# Conflicts:
#	pnpm-lock.yaml

asoee committed:
better resolution of wasm files
update from main

AndreasArvidsson commented:
@asoee Just want to get in touch and check if this is something you are still using and/or working on? If so it would probably useful if we could finalize this pull request.
